client/landing/stepper/hooks/use-update-flow-progress.ts

```diff
@@ -0,0 +1,146 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { Onboard } from '@automattic/data-stores';
+import { useIsEnglishLocale } from '@automattic/i18n-utils';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { ONBOARD_STORE } from '../stores';
+import type { StepPath } from '../declarative-flow/internals/steps-repository';
+
+const SiteIntent = Onboard.SiteIntent;
+const MAX_STEPS = 10;
+
+export function useUpdateFlowProgress( currentStep: StepPath ) {
+	const intent = useSelect( ( select ) => select( ONBOARD_STORE ).getIntent() );
```

Comment:
Can we pass `intent` as the hook's function argument? This way, the hook won't have any Redux dependencies.